README.md

```diff
@@ -1277,6 +1277,171 @@ $providerData->modifiedAt; //-> 2014-04-01T17:00:09.189Z
+collection of their API Keys. There will also be a list of API keys on a account’s profile in the 
+Stormpath Admin Console. You will be able to both create and manage keys in both.
+
+#### CREATING API KEYS FOR AN ACCOUNT 
+
+```php
+$apiKey = $account->createApiKey();
+
+$apiKeyId = $apikey->id;
+$apiKeySecret = $apikey->secret;
+```
+
+The `ApiKey` returned will have the following properties:
+* id	    The unique identifier for the API Key
+* secret	The secret for the API key.
+* status	A property that represent the status of the key. Keys with a disabled status will not be able to authenticate.
+* account	A link to the ApiKey’s account.
+* tenant	A link to the ApiKey’s tenant.
+
+#### MANAGE API KEYS FOR AN ACCOUNT
+
+##### Deleting an API Key
+
+```php
+$apiKeyId = 'FURThLWDE4MElDTFVUMDNDTzpQSHozZ';
+$apiKey = $application->getApiKey($apiKeyId);
+$apiKey->delete();
+```
+
+##### Disable an API key
+
+```php
+$apiKeyId = 'FURThLWDE4MElDTFVUMDNDTzpQSHozZ' 
+$apiKey = $application->getApiKey('FURThLWDE4MElDTFVUMDNDTzpQSHozZ');
+$apiKey->status = 'DISABLED';
+$apiKey->save()
+```
+
+### Using the Stormpath SDK to Authenticate and Generate Tokens for your API Keys
+
+The class `ApiRequestAuthenticator` can be used to perform both Basic Authentication 
+and OAuth Authentication.
+ 
+### Basic Authentication
+
+```
+GET /troopers/tk421/equipment 
+Accept: application/json
+Authorization: Basic MzRVU1BWVUFURThLWDE4MElDTFVUMDNDTzpQSHozZitnMzNiNFpHc1R3dEtOQ2h0NzhBejNpSjdwWTIwREo5N0R2L1g4
+Host: api.trooperapp.com
+```
+
+The `Authorization` header contains a base64 encoding of the API Key and Secret.
+
+#### Using `ApiRequestAuthenticator`
+
+```php
+$application = \Stormpath\Resource\Application::get("https://api.stormpath.com/v1/applications/24mp4us71ntza6lBwlu");
+
+$request = \Stormpath\Authc\Api\Request::createFromGlobals();
+$result = new ApiRequestAuthenticator($application)->authenticate($request);
+
+$account = $result->account;
+$apiKey = $result->apiKey;
+```
+
+##### Using `BasicRequestAuthenticator`
+
+```php
+$application = \Stormpath\Resource\Application::get("https://api.stormpath.com/v1/applications/24mp4us71ntza6lBwlu");
+
+$request = \Stormpath\Authc\Api\Request::createFromGlobals();
+$result = new BasicRequestAuthenticator($application)->authenticate($request);
+
+$account = $result->account;
+$apiKey = $result->apiKey;
+```
+
+### OAuth Authentication
+
+#### GENERATING A TOKEN
+
+```
+POST /oauth/token
+Accept: application/json
+Authorization: Basic MzRVU1BWVUFURThLWDE4MElDTFVUMDNDTzpQSHozZitnMzNiNFpHc1
+Content-Type: application/x-www-form-urlencoded
+Host: api.trooperapp.com
+
+  grant_type=client_credentials
+```
+
+The `Authorization` header contains a base64 encoding of the API Key and Secret.
+
+```php
+$application = \Stormpath\Resource\Application::get("https://api.stormpath.com/v1/applications/24mp4us71ntza6lBwlu");
+
+$request = \Stormpath\Authc\Api\Request::createFromGlobals();
```

Comment:
Is there any other way for the user to create the request that is more flexible?